idaes/core/scaling/tests/test_util.py

```diff
@@ -1136,6 +1137,52 @@ def test_set_scaling_factor_overwrite_false(self, caplog):
         )
         assert m.scaling_factor[m.v] == 10
 
+    @pytest.mark.unit
+    def test_set_scaling_factor_transformed_constraint(self):
+        m = ConcreteModel()
+        m.x = Var(initialize=500)
+        m.c1 = Constraint(expr=m.x <= 1e3)
+        m.c2 = Constraint(expr=m.x == 1e3)
+        m.c3 = Constraint(expr=m.x >= 1e3)
+
+        def indexed_constraint_rule(b, idx):
+            return b.x == 1e3
+
+        m.c4 = Constraint([1, 2, 3], rule=indexed_constraint_rule)
+
+        def match(cname):
+            return re.escape(
+                f"Attempted to set constraint scaling factor for transformed constraint {cname}. "
+                "Please use only one of set_scaling_factor and constraint_scaling_transform "
+                "per constraint to avoid double scaling."
+            )
+
+        constraint_scaling_transform(m.c1, 1e-3)
+        constraint_scaling_transform(m.c2, 1e-3)
+        constraint_scaling_transform(m.c3, 1e-3)
+        for idx in m.c4:
+            constraint_scaling_transform(m.c4[idx], 1e-3)
+
+        with pytest.raises(RuntimeError, match=match("c1")):
+            set_scaling_factor(m.c1, 1e-3)
+        with pytest.raises(RuntimeError, match=match("c2")):
+            set_scaling_factor(m.c2, 1e-3)
+        with pytest.raises(RuntimeError, match=match("c3")):
+            set_scaling_factor(m.c3, 1e-3)
+        with pytest.raises(
```

Comment:
Do you have a test of the correct way to apply scaling to `c4`?

Reply:
I'm afraid I don't follow. I test applying a scaling factor both to the constraint as a whole as well as the individual indices. What other case do you want to test?

Reply:
Okay, I see now that the allowed case is also being tested.